lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -1033,6 +1021,14 @@ const char16* WasmBinaryReader::ReadInlineName(uint32& length, uint32& nameLengt
         AssertMsg(UNREACHED, "We calculated the length before decoding, what happened ?");
         ThrowDecodingError(_u("Error while decoding utf8 string"));
     }
+    for (size_t i = 0; i < decodedLength; ++i)
+    {
+        const char16 c = contents[i];
+        if (!utf8::IsValidWideChar(c) || c == UNICODE_UNKNOWN_CHAR_MARK)
+        {
+            ThrowDecodingError(_u("Invalid UTF-8 encoding"));
```

Comment:
At first I wanted the utf8 decoder to tell me if there was an invalid character, but I didn't find any non intrusive way to do it. Which is why I am checking afterward.
From what @dilijev says, I should only need to check for `UNICODE_UNKNOWN_CHAR_MARK` to see if there was any invalid characters right ?

Since strings in the wasm buffer are relatively rare (only import, export, custom section and debugging info) and small, I don't think doing another pass on the buffer will make a big difference.